src/SeqCli/Apps/Definitions/AppMetadataReader.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright 2020 Datalust Pty Ltd and Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Seq.Apps;
+
+namespace SeqCli.Apps.Definitions
+{
+    static class AppMetadataReader
+    {
+        public static AppDefinition ReadFromSeqAppType(Type seqAppType)
+        {
+            if (seqAppType == null) throw new ArgumentNullException(nameof(seqAppType));
+
+            var declared = seqAppType.GetCustomAttribute<SeqAppAttribute>();
+            if (declared == null)
+                throw new ArgumentException($"The provided type '{seqAppType}' is not marked with [SeqApp].");
+
+            var app = new AppDefinition
+            {
+                Name = declared.Name,
+                Description = declared.Description,
+                AllowReprocessing = declared.AllowReprocessing,
+                Settings = GetAvailableSettings(seqAppType),
+                Capabilities = GetCapabilities(seqAppType),
+                Platform = new Dictionary<string, AppPlatformDefinition>
+                {
+                    ["hosted-dotnet"] = new AppPlatformDefinition
```

Comment:
This replaces the `__MainReactorTypeName` top-level property; reasonable to include here because `hosted-dotnet` is a logical alternative for other platform values (`win-x64` etc.)